src/main/java/org/isf/dicomtype/gui/DicomTypeEdit.java

```diff
@@ -198,37 +197,24 @@ private JButton getOkButton() {
 
 				dicomType.setDicomTypeDescription(descriptionTextField.getText());
 				dicomType.setDicomTypeID(codeTextField.getText());
-				boolean result;
 				if (insert) {    // inserting
 					try {
-						result = dicomTypeBrowserManager.newDicomType(dicomType);
-						if (result) {
-							fireDicomTypeInserted(dicomType);
-						}
-						if (!result) {
-							MessageDialog.error(null, "angal.common.datacouldnotbesaved.msg");
-						} else {
-							dispose();
-						}
+						dicomTypeBrowserManager.newDicomType(dicomType);
+						fireDicomTypeInserted(dicomType);
+						dispose();
 					} catch (OHServiceException e1) {
-						OHServiceExceptionUtil.showMessages(e1, DicomTypeEdit.this);
+						OHServiceExceptionUtil.showMessages(e1, this);
```

Comment:
Did you test this? I recall the owner `this` was not enough, but maybe now it is different

Reply:
Yes I tested it and it works fine.

Reply:
Ok, FWIW there are plenty of these all around GUI